app/policies/teacher_policy.rb

```diff
@@ -3,6 +3,11 @@ class TeacherPolicy < ApplicationPolicy
   def index?
     user.admin?
   end
+
+  def update?
+    user.admin? && user.grove_id == resource.grove_id
```

Comment:
@Jbern16 this might have accidentally slipped in here. I think this is probably the direction the policy should go, but there needs to be testing around it and some thought about what the appropriate check is.